plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -174,20 +174,32 @@ export function createPublishGitlabAction(options: {
         [tokenType]: token,
       });
 
-      let { id: targetNamespace } = (await client.Namespaces.show(owner)) as {
-        id: number;
-      };
+      let targetNamespaceId;
+
+      try {
+        const namespaceResponse = (await client.Namespaces.show(owner)) as {
+          id: number;
+        };
+
+        targetNamespaceId = namespaceResponse.id;
+      } catch (e) {
+        if (e.response && e.response.statusCode === 404) {
+          throw new InputError(
+            `The namespace ${owner} is not found or the user doesn't have permissions to access it`,
+          );
+        }
+      }
```

Comment:
don't forget to throw the error 😄

```suggestion
        }
        throw e;
      }
```